core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -975,6 +975,8 @@ private[spark] object Utils extends Logging {
 
   def checkHost(host: String) {
     assert(host != null && host.indexOf(':') == -1, s"Expected hostname (not IP) but got $host")
+    assert(host.matches("^[a-zA-Z0-9-.]+$"),
```

Comment:
I don't feel this is the right place to do the check, if you really like a detailed error message then you shall produce that in `RpcEndpointAddress.apply(sparkUrl: String)`, but I don't think that shall add much value for users either.

Reply:
Further more, the regex even doesn't guarantee a valid host, so it looks not very reasonable to have a check like that from any point of view.

Reply:
Yeah,  it looks not very reasonable for this check, there is no good way to do it.
I might have to close this PR